packages/commerce-sdk-react/src/hooks/ShopperProducts/query.ts

```diff
@@ -11,6 +11,24 @@ import {UseQueryOptions, UseQueryResult} from '@tanstack/react-query'
 
 type Client = ApiClients['shopperProducts']
 
+const productKeys = {
+    all: [{entity: 'product'}],
+    useProducts: (arg: Record<string, unknown>) => [{...productKeys.all[0], scope: 'list', ...arg}],
```

Comment:
What's the reason for having the query key to be structured as an array of 1 item (which is an object)?

According to the [official doc](https://react-query-v3.tanstack.com/guides/query-keys#array-keys), it's possible to mix and match the array's content. For example, it can be like: 
`['product', 'list', {...arg}]`

Reply:
If we use a multiple-items array as the query key, we can easily invalidate the queries granularly like this: 
```js
queryClient.invalidateQueries(['product']) // will invalidate both `useProduct` and `useProducts`
queryClient.invalidateQueries(['product', 'list']) // will invalidate only `useProducts`
```

Reply:
The reason comes from a TkDodo's blog post: https://tkdodo.eu/blog/leveraging-the-query-function-context#object-query-keys

The author proposes to define all keys as an array with exactly one object.

The object inside the key contains the query data returned and some high-level strings we use to define the 'entity' and 'scope' ideas. 

Using an object we don't need o to worry about the specific position of a particular key in the array. Instead, we can obtain the value of a particular key by using the object name destructuring.

And we can still use fuzzy matching query keys when we need to invalidate related queries during mutations.